lms/djangoapps/discussion/apps.py

```diff
@@ -5,13 +5,31 @@
 """
 
 from django.apps import AppConfig
+from openedx.core.djangoapps.plugins.constants import ProjectType, SettingsType, PluginURLs, PluginSettings
 
 
 class DiscussionConfig(AppConfig):
     """
     Application Configuration for Grades.
     """
+    COURSE_KEY_PATTERN = r'(?P<course_key_string>[^/+]+(/|\+)[^/+]+(/|\+)[^/?]+)'
+    COURSE_ID_PATTERN = COURSE_KEY_PATTERN.replace('course_key_string', 'course_id')
```

Comment:
Can we move these to a new openedx/core/constants.py file and update lms/envs/common.py to import from there?